javaslang/generator/Generator.scala

```diff
@@ -1265,46 +1265,40 @@ def generateMainClasses(): Unit = {
 
             ${(i == 0).gen(xs"""
               @Override
-              public <T> Tuple1<T> prepend(T value) {
+              public <T> Tuple1<T> append(T value) {
                  return new Tuple1<>(value);
               }
 
               @Override
-              public <T> Tuple1<T> append(T value) {
+              public <T> Tuple1<T> prepend(T value) {
                  return new Tuple1<>(value);
               }
             """)}
 
             ${(i > 0 && i < N).gen(xs"""
               @Override
-              public <T> Tuple${i + 1}<T, ${(1 to i).gen(j => s"T$j")(", ")}> prepend(T value) {
-                  return Tuple.of(value, ${(1 to i).gen(j => s"_$j")(", ")});
-              }
-
-              @Override
               public <T> Tuple${i + 1}<${(1 to i).gen(j => s"T$j")(", ")}, T> append(T value) {
                   return Tuple.of(${(1 to i).gen(j => s"_$j")(", ")}, value);
               }
-            """)}
 
-            ${(i == N).gen(xs"""
               @Override
-              public <T> Tuple prepend(T value) {
-                  throw new UnsupportedOperationException("Prepend to Tuple${N}");
+              public <T> Tuple${i + 1}<T, ${(1 to i).gen(j => s"T$j")(", ")}> prepend(T value) {
+                  return Tuple.of(value, ${(1 to i).gen(j => s"_$j")(", ")});
               }
+            """)}
 
+            ${(i == N).gen(xs"""
               @Override
               public <T> Tuple append(T value) {
                   throw new UnsupportedOperationException("Append to Tuple${N}");
               }
-            """)}
 
-            ${(i > 0).gen(xs"""
-              public static $generics $className<${(1 to i).gen(j => s"Seq<? extends T$j>")(", ")}> sequence(Iterable<$className<${(1 to i).gen(j => s"? extends T$j")(", ")}>> tuples) {
-                Objects.requireNonNull(tuples, "tuples is null");
-                return new $className<>(${(1 to i).gen(j => s"${im.getType("javaslang.collection.Iterator")}.ofAll(tuples).map($className::_$j).toList()")(", ")});
+              @Override
+              public <T> Tuple prepend(T value) {
+                  throw new UnsupportedOperationException("Prepend to Tuple${N}");
```

Comment:
> I don't know if these methods are used often in practice, haven't seen a use-case yet. So we should consider to remove them completely. Keep things simple! Less is more.

See [this](https://github.com/javaslang/javaslang-jackson/blob/master/src/main/java/javaslang/jackson/datatype/deserialize/TupleDeserializer.java#L70) example